code/ATMOSPHERICS/components/unary_devices/vent_scrubber.dm

```diff
@@ -250,14 +250,13 @@
 				filtered_out.carbon_dioxide = removed.carbon_dioxide
 				removed.carbon_dioxide = 0
 
-			if(removed.trace_gases.len>0)
-				for(var/datum/gas/trace_gas in removed.trace_gases)
-					if(istype(trace_gas, /datum/gas/oxygen_agent_b))
-						removed.trace_gases -= trace_gas
-						filtered_out.trace_gases += trace_gas
-					else if(istype(trace_gas, /datum/gas/sleeping_agent) && scrub_N2O)
-						removed.trace_gases -= trace_gas
-						filtered_out.trace_gases += trace_gas
+			if(removed.agent_b)
+				filtered_out.agent_b = removed.agent_b
+				removed.agent_b = 0
+
+			if(scrub_N2O)
+				filtered_out.sleeping_agent = removed.sleeping_agent
+				removed.sleeping_agent = 0
```

Comment:
Is `volatile_fuel` intentionally omitted here? (I generally see a pattern of its omission throughout the code here that doesn't appear to be corrected)

Reply:
I just put it in where it already existed, but yeah, the thought did occur to me. 

Any other locations that'd be ideal?

Reply:
I went ahead and just purged volatile fuel.

After doing some code digging, it was originally a gas that was created by splashing welding fuel on a turf--it was, predictably, horrifically overpowered.

It hasn't been used, and probably never will be---Agent B is *barely* used, but has an interesting enough reaction that I'm willing to keep it in---unless you want me to remove that as well.